internal/server/async_tcp.go

```diff
@@ -286,13 +293,62 @@ func (s *AsyncServer) executeCommandToBuffer(redisCmd *cmd.RedisCmd, buf *bytes.
 		Client:   c,
 	}
 
-	response := <-s.ioChan
-	buf.Write(response.Result)
+	resp := <-s.ioChan
+	buf.Write(resp.Result)
+}
+
+func readCommands(c io.ReadWriter) (cmd.RedisCmds, bool, error) {
+	var hasABORT = false
+	rp := response.NewRESPParser(c)
+	values, err := rp.DecodeMultiple()
+	if err != nil {
+		return nil, false, err
+	}
+
+	var cmds = make([]*cmd.RedisCmd, 0)
+	for _, value := range values {
+		arrayValue, ok := value.([]interface{})
+		if !ok {
+			return nil, false, fmt.Errorf("expected array, got %T", value)
+		}
+
+		tokens, err := toArrayString(arrayValue)
+		if err != nil {
+			return nil, false, err
+		}
+
+		if len(tokens) == 0 {
+			return nil, false, fmt.Errorf("empty command")
+		}
+
+		command := strings.ToUpper(tokens[0])
+		cmds = append(cmds, &cmd.RedisCmd{
+			Cmd:  command,
+			Args: tokens[1:],
+		})
+
+		if command == "ABORT" {
```

Comment:
nit: Shall we extract this as const command as used at multiple places?

Reply:
Can i take these changes while working on the worker/multi-threading part? I wanted to limit the refactoring to code organization.

Reply:
Sure just `nit`

Reply:
thanks